sentry_sdk/worker.py

```diff
@@ -89,7 +89,7 @@ def kill(self):
         logger.debug("background worker got kill request")
         with self._lock:
             if self._thread:
-                self._queue.put_nowait(_TERMINATOR)
+                self._queue.put(_TERMINATOR)
```

Comment:
Not sure about this part. I think it's legit to block a little more on flush/close, even beyond the timeout, if the queue is completely full.

Reply:
(sorry, fat fingered the resolve button on my phone)

Other SDKs have a timeout on operations like this. Is kill used like "close"?

Reply:
`kill` is an internal method that is supposed to be nonblocking. `close` calls `flush` with a timeout and then `kill`. I will actually revert this change for now.